File: app/src/main/java/com/zeyad/rxredux/snackbar/ColoredSnackBar.java
package com.zeyad.rxredux.snackbar;

import android.support.annotation.ColorInt;
import android.support.design.widget.Snackbar;
import android.view.View;

/**
 * @author by ZIaDo on 7/21/17.
 */
public final class ColoredSnackBar {
    private ColoredSnackBar() {
    }

    private static View getSnackBarLayout(Snackbar snackbar) {
        return snackbar != null ? snackbar.getView() : null;
    }

    private static Snackbar colorSnackBar(Snackbar snackbar, @ColorInt int colorId) {
        View snackBarView = getSnackBarLayout(snackbar);
        if (snackBarView != null) {
            snackBarView.setBackgroundColor(colorId);
        }
        return snackbar;
    }

    public static Snackbar info(Snackbar snackbar, int color) {
        return colorSnackBar(snackbar, color);
    }

    public static Snackbar error(Snackbar snackbar, int color) {
        return colorSnackBar(snackbar, color);
    }
}

File: app/src/main/java/com/zeyad/rxredux/snackbar/ColoredSnackBar.kt
package com.zeyad.rxredux.snackbar

import android.view.View
import androidx.annotation.ColorInt
import com.google.android.material.snackbar.Snackbar

object ColoredSnackBar {

    private fun getSnackBarLayout(snackBar: Snackbar): View? = snackBar.view

    private fun colorSnackBar(snackBar: Snackbar, @ColorInt colorId: Int): Snackbar =
            snackBar.apply { getSnackBarLayout(snackBar)?.setBackgroundColor(colorId) }

    fun info(snackBar: Snackbar, color: Int): Snackbar = colorSnackBar(snackBar, color)

    fun error(snackBar: Snackbar, color: Int): Snackbar = colorSnackBar(snackBar, color)
}
